### PR TITLE
Update modules to support riscv64

### PR DESCRIPTION
**Pull request by carlosedp**
Signed-off-by: CarlosEDP <me@carlosedp.com>

Updating projects using libraries that were recently updated to support Risc-V architecture.

Go upstream work is tracked on: golang/go#27532

Risc-V software support tracker on https://github.com/carlosedp/riscv-bringup

### CODE REVIEW

**Comment by tiborvass** [comment]
@carlosedp you inadvertently added the dockerd binary in the PR.

**Comment by carlosedp** [comment]
My bad, it fell into the cracks together with all the files. Should be removed now.

**Comment by tiborvass** [comment]
@carlosedp the vendor check failed, you've got to run it this way:
```
$ make BIND_DIR=. shell
# vndr
```

**Comment by carlosedp** [comment]
@tiborvass Why when I deleted whole `vendor` dir and ran `vndr` it didn't do the correct thing?

When I ran inside the shell, the difference was that some go.mod files were deleted.

**Comment by tiborvass** [comment]
@carlosedp probably vndr version differences, that's why we pin the vndr version in the dockerfile so that everyone agrees on how to vendor.

**Comment by thaJeztah** [comment]
reminds me that we need to debug why updating to the latest vndr version causes CI to fail; https://github.com/moby/moby/pull/39216

**Comment by thaJeztah** [comment]
oh; before we merge this one, could we merge https://github.com/moby/moby/pull/39054 ? in case we want to backport that one (without introducing the new things from the other bumps)

we also might want to ask if they can tag a new version for netlink

**Comment by tiborvass** [comment]
Let's also get https://github.com/moby/moby/pull/39216 first. I can rebase this PR then.

File: vendor.conf [hunk 1]
```diff
@@ -62,8 +62,8 @@ github.com/coreos/go-semver                         8ab6407b697782a06568d4b7f1db
 github.com/ugorji/go                                b4c50a2b199d93b13dc15e78929cfb23bfdf21ab # v1.1.1
 github.com/hashicorp/consul                         9a9cc9341bb487651a0399e3fc5e1e8a42e62dd9 # v0.5.2
 github.com/miekg/dns                                e57bf427e68187a27e22adceac868350d7a7079b # v1.0.7
-github.com/ishidawataru/sctp                        07191f837fedd2f13d1ec7b5f885f0f3ec54b1cb
-go.etcd.io/bbolt                                    7ee3ded59d4835e10f3e7d0f7603c42aa5e83820 # v1.3.1-etcd.8
+github.com/ishidawataru/sctp                        6e2cb1366111dcf547c13531e3a263a067715847
```
Comment (tiborvass) [inline vendor.conf]: @carlosedp why is upgrading sctp needed?

**Comment by tiborvass** [comment]
@carlosedp I updated your PR. One notable difference is that the netns commit was 13995c7128ccc8e51e9a6bd2b551020a27180abd which didn't include the riscv changes from https://github.com/vishvananda/netns/pull/34. So I updated it to 7109fa855b0ff1ebef7fbd2f6aa613e8db7cfbc0.

**Comment by carlosedp** [comment]
Amazing. Thanks @tiborvass. Now the vndr I used standalone would work fine since the CI have been updated as well right?

File: vendor.conf (hunk 1 above)
Comment (carlosedp) [reply vendor.conf]: I believe libnetwork requires it.

**Comment by tiborvass** [comment]
@carlosedp most likely, but I'd still recommend using `make BIND_DIR=. shell` just in case. Either way, the vendor CI will catch any mismatches.

For the record, I couldn't figure out this: https://github.com/moby/moby/pull/39423#discussion_r298806797

**Comment by carlosedp** [comment]
After updating libnetwork it failed building. I saw the latest libnetwork update required a new sctp too.

**Comment by tiborvass** [comment]
Ugh, forgot to include https://github.com/docker/libnetwork/pull/2412 so I updated libnetwork as well.

**Comment by thaJeztah** [comment]
vendor check is failing @tiborvass

**Comment by tiborvass** [comment]
@thaJeztah what are you talking about? 👀 😛jk updated

**Review by thaJeztah — APPROVED** [review]
LGTM

**Review by tiborvass — APPROVED** [review]

**Comment by crosbymichael** [comment]
LGTM

**Pull request merged**